src/config.rs

```diff
@@ -211,10 +213,34 @@ impl ConfigHelpItem {
     }
 }
 
+/// This is used by Config to track which config parameters are accessed during
+/// formatting. It uses a RefCell for interior mutability, as we don't want to
+/// require a mutable reference to Config in order to access configuration.
+#[derive(Clone, Default)]
+struct ConfigTracker {
```

Comment:
This struct seems unneccessary, could you just put the RefCell directly in `Config`?